packages/web/src/components/modals/TransactionInfoModals/TransactionSwap/TransactionSwap.tsx

```diff
@@ -0,0 +1,370 @@
+import type { FunctionComponent } from 'react';
+import { useEffect, useState } from 'react';
+
+import { styled } from '@linaria/react';
+import { useConnectionContext, useTransaction, useWallet } from '@p2p-wallet-web/core';
+import classNames from 'classnames';
+
+import type { ModalPropsType } from 'app/contexts';
+import { ToastManager } from 'components/common/ToastManager';
+import { Button } from 'components/ui';
+import { trackEvent } from 'utils/analytics';
+import { getExplorerUrl } from 'utils/connection';
+import { transferNotification } from 'utils/transactionNotifications';
+
+import {
+  ButtonExplorer,
+  CheckmarkIcon,
+  CloseIcon,
+  CloseWrapper,
+  Content,
+  Desc,
+  FieldsWrapper,
+  FieldTitle,
+  FieldValue,
+  FieldWrapper,
+  Footer,
+  OtherIcon,
+  ShareIcon,
+  ShareWrapper,
+  Title,
+  Wrapper,
+} from '../common/styled';
+import type { TransferParams } from '../TransactionStatusModal/Send';
+import type { SwapParams } from '../TransactionStatusModal/Swap';
+import { Swap } from '../TransactionStatusModal/Swap';
+
+/* eslint-disable */
+/* THIS FILE IS A COPY-PASTE, WILL BE REMOVED WITH
+ * https://p2pvalidator.atlassian.net/browse/PWN-3322
+ */
+
+const INITIAL_PROGRESS = 5;
+
+export const Header = styled.div`
+  position: relative;
+
+  padding: 26px 20px 50px;
+
+  text-align: center;
+`;
+
+export const ProgressWrapper = styled.div`
+  height: 1px;
+
+  background: rgba(0, 0, 0, 0.05);
+`;
+
+export const BlockWrapper = styled.div`
+  position: absolute;
+  bottom: -28px;
+  left: 50%;
+
+  display: flex;
+  align-items: center;
+
+  justify-content: center;
+  width: 55px;
+  height: 55px;
+  margin-left: -27px;
+
+  background: #5887ff;
+  border-radius: 12px;
+  &.isProcessing {
+    background: #ffd177;
+  }
+  &.isSuccess {
+    background: #77db7c;
+  }
+  &.isError {
+    background: #f77;
+  }
+`;
+
+const ProgressLine = styled.div`
+  width: ${INITIAL_PROGRESS}%;
+  height: 1px;
+
+  background: #5887ff;
+
+  transition: width 0.15s;
+`;
+
+const handleCopyClick = (str: string) => () => {
+  try {
+    void navigator.clipboard.writeText(str);
+    ToastManager.info('Copied to buffer!');
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+const DEFAULT_TRANSACTION_ERROR = 'Transaction error';
+
+type SendActionType = () => Promise<string>;
+type SwapActionType = () => Promise<string>;
+
+export type TransactionStatusModalProps = {
+  type: 'send' | 'swap';
```

Comment:
I think you know that if it modal for Swap so it should have only one type and conditions in code must be removed. 
It's just notice